src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -219,7 +219,17 @@ Shard::open(Scheduler& scheduler, nudb::context& ctx)
             std::string("exception ") + e.what() + " in function " + __func__);
     }
 
-    setBackendCache(lock);
+    // Set backend caches
+    {
+        auto const size{config.getValueFor(SizedItem::nodeCacheSize, 0)};
```

Comment:
I get why you chose `0` here but is auto-sizing to `tiny` a good idea here? The caches would end up up holding ~16k entries. Perhaps having a more intelligent scheme could afford better usage of resources; for example, if the configured `NODE_SIZE` is `large` or `huge` we configure these caches to be `medium`, otherwise we opt for `tiny`.

Reply:
A shard's cache can be much smaller than the node store cache, especially completed shards. A possible optimization is to use a larger cache size with the 'current' shard as it will need to have better performance. We can easily add that when we offer the ability to use shards as the primary store.